templates/deployment.yaml

```
@@ -64,8 +64,12 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       {{- if .Values.serviceAccount.enabled }}
+      {{- if .Values.serviceAccount.name}}
+      serviceAccountName: {{ .Values.serviceAccount.name }}
+      {{- else }}
       serviceAccountName: {{ .Release.Namespace }}
       {{- end }}
```

If we were to use the [default function](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function), this could be simplified a tad: 
```suggestion
      serviceAccountName: {{ .Values.serviceAccount.name | default .Release.Namespace }}
```